core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -811,18 +795,31 @@ object IO {
         } yield fiberA.zipWith(fiberAs)(_ :: _)
     }
 
+  final def bracket0[E, A, B](
+    acquire: IO[E, A]
+  )(release: (A, ExitResult[E, B]) => Infallible[Unit])(use: A => IO[E, B]): IO[E, B] =
+    IORef[E, Option[(A, ExitResult[E, B])]](None).flatMap { m =>
+      (for {
+        a <- acquire.flatMap(a => m.write[E](Some((a, ExitResult.Terminated(new Throwable)))).const(a)).uninterruptibly
```

Comment:
Probably best to use an error message here in the `Throwable`, like "Interrupted fiber".

It bugs me we can't get the inner `t` out, but this is the best we can do for now.

Reply:
Yep, had to dash off right after. It would be nice to have an ternary `Option[A, B]` sometimes.